api/v1alpha8/openstackcluster_types.go

```diff
@@ -34,10 +34,12 @@ type OpenStackClusterSpec struct {
 	// +optional
 	CloudName string `json:"cloudName"`
 
-	// NodeCIDR is the OpenStack Subnet to be created. Cluster actuator will create a
-	// network, a subnet with NodeCIDR, and a router connected to this subnet.
+	// NodeSubnets are describing OpenStack Subnets to be created. Cluster actuator
+	// will create a network, subnets with NodeCIDR, and a router connected to these subnets.
+	// Currently only one IPv4 subnet is supported.
 	// If you leave this empty, no network will be created.
-	NodeCIDR string `json:"nodeCidr,omitempty"`
+	// +kubebuilder:validation:MaxItems=1
+	NodeSubnets []SubnetSpec `json:"nodeSubnets,omitempty"`
 
 	// If NodeCIDR is set this option can be used to detect an existing router.
```

Comment:
We need to update these comments where NodeCIDR is mentioned.

Also re-reviewing this, I realized that we also have `Network`, `Subnets`, `Router`.
Maybe to avoid confusion we could rename `NodeSubnets` to `ManagedSubnets` in the same vain as `ManagedSecurityGroups`  (see [PR](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1826)).

Reply:
Yeah, that's not a bad idea. I've applied it, let's see if correctly.